test/e2e/apps/statefulset.go

```diff
@@ -510,6 +512,44 @@ var _ = SIGDescribe("StatefulSet", func() {
 
 		})
 
+		ginkgo.It("should perform canary updates and phased rolling updates of template modifications for partiton1 and delete pod-0", func(ctx context.Context) {
+			ginkgo.By("Creating a new StatefulSet without failing container")
+			ss := e2estatefulset.NewStatefulSet("ss2", ns, headlessSvcName, 3, nil, nil, labels)
+			deletingPodForRollingUpdatePartitionTest(ctx, c, ns, ss)
+
+			ginkgo.By("Creating a new StatefulSet with failing container")
```

Comment:
I would suggest to use dedicated `ginkgo.It` for the test case to make sure the test cases are truly independent.